### ISSUE TITLE
Panic lint doesn't catch panics with formatting

### ISSUE THREAD

**Issue by paulkernfeld** [rust-lang/rust-clippy#5767]
It looks to me like the `panic` lint doesn't catch panics with formatting. Although maybe this is intentional, I would expect it to treat panics with and without formatting the same way.

I tried this code:

```rust
#[deny(clippy::panic)]
pub fn main() {
    panic!("{}", "This is a panic");
}
```

I expected to `cargo clippy` to fail on this code, but it passes.

### Meta

- `cargo clippy -V`: clippy 0.0.212 (bb37a0f9 2020-06-16)
- `rustc -Vv`:

```
rustc 1.44.1 (c7087fe00 2020-06-17)
binary: rustc
commit-hash: c7087fe00d2ba919df1d813c040a5d47e43b0fe7
commit-date: 2020-06-17
host: x86_64-apple-darwin
release: 1.44.1
LLVM version: 9.0
```

**Comment by JarredAllen** [comment]
I should be able to fix this.

**Comment by flip1995** [comment]
This might be related to https://github.com/rust-lang/rust-clippy/issues/3528

**Comment by JarredAllen** [comment]
I believe this can be closed, as my PR resolved it.
